ci/meson/meson_setup_phases.py

```
@@ -527,60 +529,93 @@ def detect_test_file_changes(
-        return False
+        try:
+            # save_cache() writes UTF-8; reading with the platform locale codec
+            # (cp1252 on Windows) turns any non-ASCII example path or @filter
+            # string into a permanent decode failure, and therefore a
+            # reconfigure on every single run.
+            with open(example_cache_file, "r", encoding="utf-8") as f:
+                cache_data = json.load(f)
+            cached_example_hash = cache_data.get("hash", "")
+        except KeyboardInterrupt as ki:
+            handle_keyboard_interrupt(ki)
+        except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)